### PR TITLE
create event of inventory that sync with Devices

### PR DESCRIPTION
**Pull request by mgold1234**
add unit test as well
realted to story THEEDGE-2727

Signed-off-by: mgold1234 <mgold@redhat.com>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/services/device/event_inventory_sync_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+						LastHandleTime: time.Now().Format(time.RFC3339),
+						RequestID:      faker.UUIDHyphenated(),
+					}
+					Expect(edgePayload).ToNot(BeNil())
+
+					event := &eventReq.EventInventorySyncHandler{}
+					event.RedHatOrgID = ""
+					event.Data = *edgePayload
+					mockDeviceService.EXPECT().SyncInventoryWithDevices(event.RedHatOrgID)
+					event.Consume(ctx)
```
Comment (tomershinhar) [inline pkg/services/device/event_inventory_sync_test.go:68]: where are you validating an error here?
how can we tell it did not work?

**Review by tomershinhar — COMMENTED** [review]
added a comment

File: pkg/services/devices/event_inventory_sync_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+					}
+					event := &eventReq.EventInventorySyncHandler{}
+					event.RedHatOrgID = ident.Identity.OrgID
+					event.Data = *edgePayload
+					mockDeviceService.EXPECT().SyncInventoryWithDevices(event.RedHatOrgID).Return()
+					event.Consume(ctx)
+				})
+			})
+			Context("inventory sync is requested when orgID is empty", func() {
+				It("should not be ok", func() {
```
Comment (ldjebran) [inline pkg/services/devices/event_inventory_sync_test.go:53]: it should not be OK

File: pkg/services/devices/event_inventory_sync_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+						Identity:       ident,
+						LastHandleTime: time.Now().Format(time.RFC3339),
+						RequestID:      faker.UUIDHyphenated(),
+					}
+					Expect(edgePayload).ToNot(BeNil())
+
+					event := &eventReq.EventInventorySyncHandler{}
+					event.RedHatOrgID = ""
+					event.Data = *edgePayload
+					mockDeviceService.EXPECT().SyncInventoryWithDevices(event.RedHatOrgID)
```
Comment (ldjebran) [inline pkg/services/devices/event_inventory_sync_test.go:67]: it should not be OK, but SyncInventoryWithDevices was called , something is wrong here

File: pkg/services/devices/event_inventory_sync_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+					edgePayload := &models.EdgeBasePayload{
+						Identity:       ident,
+						LastHandleTime: time.Now().Format(time.RFC3339),
+						RequestID:      faker.UUIDHyphenated(),
+					}
+					Expect(edgePayload).ToNot(BeNil())
+
+					event := &eventReq.EventInventorySyncHandler{}
+					event.RedHatOrgID = ""
```
Comment (ldjebran) [inline pkg/services/devices/event_inventory_sync_test.go:65]: it should not be ok for any orgID different from the one in identity

**Comment by ldjebran** [comment]
Please take a look at this fix https://github.com/RedHatInsights/edge-api/pull/1709  to understand the problem.

as in this PR mockDeviceService.EXPECT().SyncInventoryWithDevices(event.RedHatOrgID) has no function , it can be added or removed and the test will pass

File: pkg/services/devices/event_inventory_sync_test.go (hunk 3 above)
Comment (mgold1234) [reply pkg/services/devices/event_inventory_sync_test.go]: @ldjebran change it, thank you

File: pkg/services/devices/event_inventory_sync_test.go (hunk 4 above)
Comment (mgold1234) [reply pkg/services/devices/event_inventory_sync_test.go:74]: @ldjebran change it, thank you

**Comment by mgold1234** [comment]
image

**Review by ldjebran — APPROVED** [review]
LGTM

**Review by tomershinhar — APPROVED** [review]
LGTM

**Pull request merged**